### PR TITLE
feat: Switch all validators to injectable constructors

### PR DESCRIPTION
**Pull request by aababilov**
Injectable constructors provide a cleaner interface for dependency injection than injectable fields.

### CODE REVIEW

**Comment by aababilov** [comment]
This depends on https://github.com/MobilityData/gtfs-validator/pull/814

**Comment by aababilov** [comment]
@barbeau PTAL. All validators are updated.

I also fixed many tests that were not asserting anything due to the wrong `()` placement.

No op:
```
assertThat(noticeContainer.getValidationNotices().isEmpty())
```

Correct code:
```
assertThat(noticeContainer.getValidationNotices()).isEmpty()
```

**Comment by lionel-nj** [comment]
> I also fixed many tests that were not asserting anything due to the wrong () placement.

Thanks for doing that.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java [hunk 1]
```diff
@@ -42,8 +42,12 @@
  */
 @GtfsValidator
 public class AgencyConsistencyValidator extends FileValidator {
+  @Inject
```
Comment (lionel-nj) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java:45]:
```suggestion
  private final GtfsAgencyTableContainer agencyTable;
  
  @Inject
```

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java [hunk 2]
```diff
@@ -42,8 +42,12 @@
...
+  AgencyConsistencyValidator(GtfsAgencyTableContainer agencyTable) {
+    this.agencyTable = agencyTable;
+  }
 
-  @Inject GtfsAgencyTableContainer agencyTable;
+  private final GtfsAgencyTableContainer agencyTable;
```
Comment (lionel-nj) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java:50]:
```suggestion
```

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java (hunk 1 above)
Comment (lionel-nj) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java:45]: I would declare agencyTable before injecting the constructor to keep consistency with the other validators.

**Review by lionel-nj — CHANGES_REQUESTED** [review]
Thanks @aababilov. One small change to keep consistency, details inline.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java (hunk 1 above)
Comment (aababilov) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java:45]: Agree, done.

**Comment by aababilov** [comment]
My pleasure! Thanks for review.

**Review by lionel-nj — APPROVED** [review]
LGTM! Thanks @aababilov, there are some conflicts to solve before we merge this PR.

**Comment by aababilov** [comment]
Solved!

Do you mind to approve and merge https://github.com/MobilityData/gtfs-validator/pull/814?

**Comment by lionel-nj** [comment]
Done :)

**Comment by aababilov** [comment]
Merci beaucoup, Lionel !

Den tors 18 mars 2021 06:59Lionel Nébot Janvier ***@***.***>
skrev:

**Pull request merged**